torch/_export/exported_program.py

```diff
@@ -164,10 +191,106 @@ def transform(self, *passes: PassType) -> "ExportedProgram":
         )
         return transformed_ep
 
-    def _add_runtime_assertions(self) -> "ExportedProgram":
-        return self.transform(
-            _AddRuntimeAssertionsForConstraintsPass(self.range_constraints, self.equality_constraints)
+    def _add_runtime_assertions(
+        self, functionalize_assertions: bool,
+    ) -> "ExportedProgram":
+        ep = self.transform(
+            _AddRuntimeAssertionsForConstraintsPass(
+                self.range_constraints,
+                self.equality_constraints,
+            ),
+        )
+        if functionalize_assertions:
+            # Ideally `_update_after_adding_runtime_assertions` should run
+            # whenever `_AddRuntimeAssertionsForConstraintsPass` runs, here
+            # bundle it with `functionalize_assertions` for safety.
+            ep = _update_after_adding_runtime_assertions(self, ep)
+            ep = _update_after_functionalizing_runtime_assertions(
+                ep,
+                _functionalize_side_effectful_ops(gm=ep.graph_module),
```

Comment:
If you make `_functionalize_side_effectful_ops` similar to `_AddRuntimeAssertionsForConstraintsPass`, then you can just have utility function that updates the ExportedProgram signatures.

Reply:
So I think:
* Ideally for each pass, the signature update should follow graph transformation.
* However, I don't see easy way to implement the above using fx `PassManager`.
* A better solution I think is to just create a `ExportPassManager` which allowing running signature update after each pass run.

But I think this could be a separate PR and goes beyond the scope? But anyway, let me at least add `update_exported_program_signature ` as a class instance method of base class `ExportPassBase` so each concrete could override it (even we need to call it manually for now).